pkg/cache/tas_flavor_snapshot.go

```diff
@@ -296,6 +297,22 @@ func levelKey(topologyRequest *kueue.PodSetTopologyRequest) *string {
 	return nil
 }
 
+// findMostAllocatedDomainIdx binsearches a sorted array of domains and finds index of
+// the first domain with the lowest value of state, higher or equal than count.
+// If such a domain doesn't exist, it returns 0 as it's an index of the domain with the
+// most available resources
+func findMostAllocatedDomainIdx(domains []*domain, count int32) int {
+	mostAllocatedFitIdx := 0
+	for i, domain := range domains {
+		if domain.state < count {
+			break
+		} else if domain.state != domains[mostAllocatedFitIdx].state {
+			mostAllocatedFitIdx = i
+		}
```

Comment:
```suggestion
		}
		if domain.state != domains[mostAllocatedFitIdx].state {
			mostAllocatedFitIdx = i
		}
```

It just simplified the logic.

Reply:
Done